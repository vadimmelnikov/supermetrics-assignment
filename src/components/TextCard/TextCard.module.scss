@import 'styles/main';

.root {
  color: var(--color-glaucous);
  border: 4px solid var(--color-glaucous);
  border-radius: 4px;
  display: block;
  margin: 0;
  padding: 0;
  width: 100%;
  transition: border-color var(--transitionTime), color var(--transitionTime);
  margin-bottom: var(--gap12);
  min-height: 170px;
}

.header {
  display: block;
  border-bottom: 4px solid currentColor;
  padding: var(--gap16);
}

.date {
  margin: 0;
  color: var(--color-black);
  font-weight: 600;
  font-size: 12px;
}


.content {
  padding: var(--gap16);
  font-size: 14px;
  line-height: 20px;
  color: var(--color-black);
  
  a {
    color: var(--color-glaucous);
  }
}