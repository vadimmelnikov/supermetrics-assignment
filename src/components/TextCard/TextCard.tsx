import { formatDate } from 'utils/dates';

import s from './TextCard.module.scss';

interface Props {
  date: Date;
  description: string;
}

const TextCard = ({ date, description }: Props) => (
  <article className={s.root}>
    <header className={s.header}>
      <h4 className={s.date}>{formatDate(date)}</h4>
    </header>
    <div className={s.content}>{description}</div>
  </article>
);

export default TextCard;
