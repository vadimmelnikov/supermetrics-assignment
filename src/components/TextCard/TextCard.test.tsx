import { render, screen } from '@testing-library/react';

import TextCard from './TextCard';

const DATE = new Date('2020-05-12T23:50:21.817Z');

const DESCRIPTION = '123123';

describe('TextCard component', () => {
  it('TextCard renders', () => {
    render(<TextCard date={DATE} description={DESCRIPTION} />);

    expect(screen.getByText('123123')).toBeInTheDocument();
  });

  it('TextCard is null', () => {
    render(<TextCard date={DATE} description={DESCRIPTION} />);

    expect(screen.queryByRole('123123')).toBeNull();
  });

  it('TextCard snapshot', () => {
    const card = render(<TextCard date={DATE} description={DESCRIPTION} />);

    expect(card).toMatchSnapshot();
  });
});
