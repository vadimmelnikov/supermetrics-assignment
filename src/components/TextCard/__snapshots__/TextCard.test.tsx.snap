// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextCard component TextCard is null 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="root"
      >
        <header
          class="header"
        >
          <h4
            class="date"
          >
            May 12, 2020 23:50:21
          </h4>
        </header>
        <div
          class="content"
        >
          123123
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="root"
    >
      <header
        class="header"
      >
        <h4
          class="date"
        >
          May 12, 2020 23:50:21
        </h4>
      </header>
      <div
        class="content"
      >
        123123
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TextCard component TextCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="root"
      >
        <header
          class="header"
        >
          <h4
            class="date"
          >
            May 12, 2020 23:50:21
          </h4>
        </header>
        <div
          class="content"
        >
          123123
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="root"
    >
      <header
        class="header"
      >
        <h4
          class="date"
        >
          May 12, 2020 23:50:21
        </h4>
      </header>
      <div
        class="content"
      >
        123123
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
